lib/proxy/request_id_middleware.rb

```diff
@@ -0,0 +1,14 @@
+module Proxy
+  class RequestIdMiddleware
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      Thread.current.thread_variable_set(:request_id, env['HTTP_X_REQUEST_ID']) if env.has_key?('HTTP_X_REQUEST_ID')
+      to_ret = @app.call(env)
+      Thread.current.thread_variable_set(:request_id, nil)
```

Comment:
This isn't going to work for logging outside of rack app: the middleware is injected after webrick passes control to it, but before the controller code is invoked. There isn't much I can about this short of changing webrick, which I'd like to avoid.

What I could do however is to log the request (probably via additional middleware) and disable debug logging of requests in webrick/rack?